template/template_process.go

```diff
@@ -7,54 +7,246 @@ import (
 	"os"
 	"path"
 	"path/filepath"
+	"regexp"
 	"strings"
-	"text/template"
 
+	"github.com/coveo/gotemplate/collections"
 	"github.com/coveo/gotemplate/errors"
 	"github.com/coveo/gotemplate/utils"
+	"github.com/fatih/color"
 	"golang.org/x/crypto/ssh/terminal"
 )
 
-var templateExt = []string{".gt", ".template"}
+var (
+	templateExt    = []string{".gt", ".template"}
+	linePrefix     = `^template: ` + p(tagFile, `.*?:`+p(tagLine, `\d+`)+`(:`+p(tagCol, `\d+`)+`)?: `)
+	execPrefix     = linePrefix + `executing ".*" at <` + p(tagCode, `.*`) + `>: `
+	templateErrors = []string{
+		execPrefix + `map has no entry for key "` + p(tagKey, `.*`) + `"$`,
+		execPrefix + `(?s)error calling (raise|assert): ` + p(tagMsg, `.*`),
+		execPrefix + p(tagErr, `.*`) + `$`,
+		linePrefix + p(tagErr, `.*`) + `$`,
+	}
+)
+
+func p(name, expr string) string { return fmt.Sprintf("(?P<%s>%s)", name, expr) }
+
+const (
+	noValue      = "<no value>"
+	noValueRepl  = "!NO_VALUE!"
+	nilValue     = "<nil>"
+	nilValueRepl = "!NIL_VALUE!"
+	undefError   = `"` + noValue + `"`
+	noValueError = "contains undefined value(s)"
+	runError     = `"<RUN_ERROR>"`
+	tagLine      = "line"
+	tagCol       = "column"
+	tagCode      = "code"
+	tagMsg       = "message"
+	tagFile      = "file"
+	tagKey       = "key"
+	tagErr       = "error"
+)
 
 // ProcessContent loads and runs the file template.
 func (t Template) ProcessContent(content, source string) (string, error) {
-	content = t.substitute(content)
-
-	if strings.HasPrefix(content, "#!") {
-		// If the content starts with a Shebang operator including gotemplate, we remove the first line
-		lines := strings.Split(content, "\n")
-		if strings.Contains(lines[0], "gotemplate") {
-			content = strings.Join(lines[1:], "\n")
-			t.options[OutputStdout] = true
+	return t.processContentInternal(content, source, nil, 0)
```

Comment:
I can't find a test for this new stuff. Seems to me like it would be worthwhile to have an automated test that shows that gotemplate can display multiple errors.

Reply:
Good point, I added some tests.